src/libtiled/mapreader.cpp

```diff
@@ -825,6 +823,30 @@ void MapReaderPrivate::readTileLayerData(TileLayer &tileLayer,
 
     mMap->setLayerDataFormat(layerDataFormat);
 
+    if (mMap->infinite()) {
+        while (xml.readNext() != QXmlStreamReader::Invalid) {
+            if (xml.isEndElement()) {
+                break;
+            } else if (xml.isStartElement()) {
+                if (xml.name() == QLatin1String("chunk")) {
+                    const QXmlStreamAttributes atts = xml.attributes();
+                    int startX = atts.value(QLatin1String("startx")).toInt();
+                    int startY = atts.value(QLatin1String("starty")).toInt();
```

Comment:
Please have attributes `x`, `y`, `width` and `height` for `chunk` elements. The file format should not rely on a hard-coded internal chunk size. If you write it in the file it is much more flexible.